### PR TITLE
Let block-upgrade continue to work

### PR DESCRIPTION
**Pull request by mpolden**
FYI @hmusum @tokle

### CODE REVIEW

**Comment by mpolden** [comment]
The previous change didn't work as expected. As
```
BlockUpgrade = element block-upgrade {
    BlockChange 
}
```

means `<block-upgrade><block-change>...</block-change></block-upgrade>`

File: config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java [hunk 1]
```diff
@@ -157,7 +156,8 @@ private DeclaredZone readDeclaredZone(Environment environment, Element regionTag
             // TODO: Remove block-upgrade on Vespa 7
             if ( ! blockChangeTag.equals(tag.getTagName()) && !"block-upgrade".equals(tag.getTagName())) continue;
 
-            boolean blockVersions = trueOrMissing(tag.getAttribute("version"));
+            boolean blockVersions = tag.getTagName().equals("block-upgrade") || //  TODO: Remove condition on Vespa 7
+                                    trueOrMissing(tag.getAttribute("version"));
```
Comment (bratseth) [inline config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java]: Why is this necessary? The "version" attribute is missing if its block-upgrade.

**Review by bratseth — COMMENTED** [review]
Don't see anything wrong with this, but I also don't see the need  :-)

**Comment by bratseth** [comment]
Ok, saw your comment. So, the schema change is needed. The code change doesn't hurt, so merging then.
